src/cpp/src/gguf_utils/gguf_modeling.cpp

```diff
@@ -0,0 +1,156 @@
+#include <map>
+#include <vector>
+#include <string>
+#include <iostream>
+#include <chrono>
+#include <stdexcept>
+
+#include <openvino/openvino.hpp>
+#include "openvino/runtime/core.hpp"
+#include "openvino/opsets/opset13.hpp"
+
+#include "building_blocks.hpp"
+#include "gguf_modeling.hpp"
+
+
+using namespace ov;
+using namespace ov::op::v13;
+using namespace ov::op;
+
+auto set_name = [](auto node, auto name) {
+        node->output(0).set_names({name});
+        node->set_friendly_name(name);
+};
+
+// Also valid for other models, e.g. SmolLMs
+std::shared_ptr<ov::Model> create_llama_model(
+    const std::map<std::string, GGUFMetaData>& configs,
+    std::unordered_map<std::string, ov::Tensor>& consts) {
+
+    auto start_time = std::chrono::high_resolution_clock::now();
+    std::cout << "Start generating OV model..." << std::endl;
```

Comment:
should we remove prints to output?

Reply:
I would keep them at this point as the feature is under development.